release/visor.js

```diff
@@ -1094,77 +1094,120 @@ if (typeof module !== 'undefined' && typeof exports !== 'undefined' && module.ex
-                                var chain = next.name.split('.');
-                                chain.pop(); //remove the leftmost
-                                var parent = chain.join('.');
-                                next = $state.get(parent);
-                            } else {
-                                next = null;
-                            }
-                        }
-                        return perms;
-                    };
-                    var $urlRouter = $injector.get('$urlRouter');
-                    var toUrl = null;
-                    var bypass = false;
-                    $rootScope.$on('$stateChangeStart', function (e, toState, toParams) {
-                        if (bypass) {
-                            bypass = false;
-                            return;
-                        }
-                        toUrl = $state.href(toState, toParams).replace(/^#/, '');
-                        var shouldContinue = visorPermissions.onRouteChange(toState, function delayChange(promise) {
-                            promise.then(function () {
-                                bypass = true;
-                                $state.go(toState, toParams);
-                            })
-                        });
-                        if (!shouldContinue || shouldContinue === 'delayed') {
-                            e.preventDefault();
-                        }
-                    });
-                    visorPermissions.invokeNotAllowed = function (notAllowed) {
-
-                        //timeout is required because when using preventDefault on $stateChangeStart, the url is
-                        //reverted to it's original location, and no change at this time will override this.
-                        $timeout(function () {
-                            $injector.invoke(notAllowed, null, {restrictedUrl: toUrl})
-                        }, 0);
-                    }
-                    visorPermissions.getRoute = function (routeId) {
-                        return $state.get(routeId);
-                    };
-                }]);
-
-            }
-        }])
-})();})(window, window.angular);
\ No newline at end of file
+(function () {
+    /**
+     * @ngdoc overview
+     * @name visor.ui-router
+     * @description
+     *
+     * # Visor.ui-router
+     *
+     * `Visor.ui-router` automatically add supports for permissions in ui-router, if ui-router exists.
+     *
+     */
+    angular.module('visor.ui-router', ['visor.permissions'])
+        .run(['$rootScope', '$transitions', 'visorPermissions', '$injector', '$timeout', function ($rootScope, $transitions, visorPermissions, $injector, $timeout) {
+            var uiModuleExists = false;
+            var uiRouterV1 = false;
+            try {
+                $injector.get('$state');
+                uiModuleExists = true;
+                $injector.get('$transitions');

```

Comment:
if $transition doesn't exists, this will fail before the function is even ran, since you added the $transitions in the function arguments injection